### PR TITLE
Initial model classes added

### LINKED ISSUE #50: Create Model Classes
**Description:** 
Create the classes
-User
-MoodEvent
-SocialSituation
-EmotionalState
-FollowRequest
-MoodEventAssociation

### PR DESCRIPTION
**Pull request by bojotamara**
Closes #50 

**Summary**
The initial model classes. This is different than our UML diagram in an effort to add more abstractions. Classes include
-User
-FollowableUser
-MoodletUser
-FollowRequest
-FollowList
-Event
-MoodEvent
-SocialSituation
-EmotionalState
-MoodEventAssociation

This doesn't have everything the classes need and can be added on to.

### CODE REVIEW

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/models/EmotionalState.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    ANGRY(),
+    EXCITED(),
+    CONFUSED(),
+    SCARED(),
+    JEALOUS();
+
+    private Color color;
+    private Bitmap emoticon;
+
+    EmotionalState() {
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/models/EmotionalState.java:18]: Should the constructor be set to private?

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/models/Event.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        // This sets the current date
+        this.date = new Date();
+    }
+    public Date getDate() {
+        return date;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
```
Comment (darrentran) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/models/Event.java:21]: Do we also need a location setter?

Comment (bojotamara) [reply Moodlet/app/src/main/java/com/cmput3owo1/moodlet/models/Event.java:21]: yeah good catch

**Review by darrentran — APPROVED** [review]
good job!

**Pull request merged**